frmts/gtiff/geotiff.cpp

```diff
@@ -23336,24 +23333,24 @@ static std::mutex oDeleteMutex;
 static TIFFCodec *pJXLCodec = nullptr;
 #endif
 
-int GTiffOneTimeInit()
+void GTiffOneTimeInit()
 
 {
     std::lock_guard<std::mutex> oLock(oDeleteMutex);
 
+    static bool bOneTimeInitDone = false;
```

Comment:
Not sure if applies in this context but did you consider using thread safe std::call_once ? https://en.cppreference.com/w/cpp/thread/call_once